packages/clerk-js/src/ui/elements/withAuthHref.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { forwardRef } from 'react';
+
+import { useCoreClerk } from '../contexts';
+
+type WithAuthHrefProps<T> = T & { href?: string; useBuildUrlWithAuth?: boolean };
```

Comment:
Why do we need the `useBuildUrlWithAuth` option here?

Reply:
We need it on links such as the `signUpUrl` on the sign in page. In general, every link should have that except for the clerk marketing page url.